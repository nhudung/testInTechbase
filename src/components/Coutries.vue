<template>
	<div class="coutries-select">
		<v-overflow-btn
			:items="data"
			label="Viet Nam"
			editable
			item-value="Slug"
			item-text="Country"
			@change="handleClick"
		></v-overflow-btn>
	</div>
</template>

<script>
	export default {
		props: {
			data: {
				type: Array,
				required: true,
			},
		},
		methods: {
			handleClick(value) {
				this.$emit("getByCountry", value);
			},
		},
	};
</script>

<style lang="scss">
	.coutries-select {
		.v-input__slot {
			border-color: transparent !important;
			box-shadow: 0 1px 6px 0 rgba(32, 33, 36, 0.28);
		}
		.v-input {
			padding: 0;
			margin: 0;
		}
	}
</style>
