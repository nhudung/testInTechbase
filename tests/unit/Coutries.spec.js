import { mount } from "@vue/test-utils";
import Coutries from "@/components/Coutries.vue";

describe("Coutries.spec.js", () => {
	let cmp;

	beforeEach(() => {
		cmp = mount(Coutries, {
			// Beaware that props is overriden using `propsData`
			propsData: {
				data: [
					{
						Country: "Israel",
						Slug: "israel",
						ISO2: "IL",
					},
				],
			},
		});
	});

	it("has received data as the message property", () => {
		expect(cmp.vm.data).toEqual([
			{
				Country: "Israel",
				Slug: "israel",
				ISO2: "IL",
			},
		]);
	});
	it("has the expected html structure", () => {
		expect(cmp.element).toMatchSnapshot();
	});
	it("is a Coutries component", () => {
		expect(cmp.is(Coutries)).toBe(true);
	});
	// Vue instance
	it("Coutries is a vue instances", () => {
		expect(cmp.isVueInstance()).toBe(true);
	});
});
