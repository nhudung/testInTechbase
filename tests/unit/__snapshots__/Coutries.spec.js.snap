// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coutries.spec.js has the expected html structure 1`] = `
<div
  class="coutries-select"
>
  <v-overflow-btn
    editable=""
    item-text="Country"
    item-value="Slug"
    items="[object Object]"
    label="Viet Nam"
  />
</div>
`;
